### ISSUE TITLE
How to reference GitHub repository on a DockerHub repository?

### ISSUE THREAD

**Issue by David-Lor** [docker/hub-feedback#2138]
DockerHub repositories that have, or had in the past, a GitHub repository associated to them so that DockerHub could automatically trigger builds when any push on the GitHub repository happens, automatically embed a "Source Repository" section, under the "Docker Pull Command" and "Owner" sections, on the right side of a image repository page.

For example, this repository of mine includes the mentioned reference: https://hub.docker.com/r/davidlor/ssh-port-forward-client

However, when no GitHub repository is associated for automated builds (which is currently not available for the dockerhub Free tier), it seems like there's no way to link the GitHub repository in any other manner, isn't it?

**Comment by mikeparker** [comment]
Yes you're right, this is a confusing gap in functionality. We kept all the existing GitHub integrations when we turned off free Autobuilds, so if users upgrade to a paid account, their builds will start working again. However, this means free users will see the GitHub link without the means of removing it. Right now there is no way for a free user to add a github integration purely for the purposes of automating the readme update or linking to the source code.

We're figuring out the future of the GitHub integration whilst we continue to help migrate users across to a paid account. If you have thoughts about improving the GitHub integration longer term and what features might be helpful, please feel free to let us know.

**Comment by David-Lor** [comment]
up

**Comment by dpprdan** [comment]
> We kept all the existing GitHub integrations when we turned off free Autobuilds, so if users upgrade to a paid account, their builds will start working again.

@mikeparker I see why you would do that as a convenience feature, but please note that it is also possible that the GitHub repo (URL) changes in the meantime and there is no way of updating it at the moment. See e.g. https://hub.docker.com/r/rocker/cuda which is now built from https://github.com/rocker-org/rocker-versioned2 but still (wrongly) links to https://github.com/rocker-org/ml as the Source Repo.

**Comment by sepulzera** [comment]
> I just started hosting some open source projects and I searched half an hour on how to set it. Never crossed my mind that you have to pay to link it.

This. +1
Please let free accounts manage links to the source repo.

**Comment by yikyo** [comment]
+1

**Comment by d4kine** [comment]
+1

**Comment by anirban37** [comment]
+1

**Comment by hanseartic** [comment]
+1

**Comment by MerliX** [comment]
+1

**Comment by derlin** [comment]
+1

**Comment by nazmang** [comment]
+1

**Comment by borrougagnou** [comment]
+1

**Comment by dpprdan** [comment]
@sepulzera, @semaja2, @yikyo, @d4kine, @anirban37, @hanseartic, @MerliX, @derlin, @nazmang, @borrougagnou, @Anarchotect, 
y'all are probably not aware, but a "+1" comment notifies all subscribers without adding any meaningful content to the discussion. This is a distraction for all involved and does not speed up the resolution of the issue in any way, so please don't do this.

Instead it's best practice to leave a 👍🏻 on the original post to show your support.

**Comment by borrougagnou** [comment]
The way how github works shows just how useless the reaction system is.

An issue comment is more visible than a useless thumbsup react smiley

**Comment by kabudu** [comment]
Bump. It doesn't make sense to have to pay for Dockerhub just to add a **link** to the source repository, without the requirement of autobuilds. In theory, I could just reference the repository in the overview, but it's not as nice.

**Comment by sepulzera** [comment]
Is this seriously still not resolved? How? oO
Bump!
